cachi2/core/package_managers/rubygems.py

```diff
@@ -0,0 +1,264 @@
+import logging
+import re
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Optional
+
+from gemlock_parser.gemfile_lock import Gem, GemfileLockParser  # type: ignore
+from packageurl import PackageURL
+
+from cachi2.core.errors import FetchError
+from cachi2.core.models.input import Request, RubygemsPackageInput
+from cachi2.core.models.output import Component, EnvironmentVariable, ProjectFile, RequestOutput
+from cachi2.core.package_managers.general import download_binary_file, extract_git_info
+from cachi2.core.rooted_path import RootedPath
+from cachi2.core.scm import clone_as_tarball
+
+GEMFILE_LOCK = "Gemfile.lock"
+
+GIT_REF_FORMAT = re.compile(r"^[a-fA-F0-9]{40}$")
+PLATFORMS_RUBY = re.compile(r"^PLATFORMS\n {2}ruby\n\n", re.MULTILINE)
+
+log = logging.getLogger(__name__)
+
+
+def fetch_rubygems_source(request: Request) -> RequestOutput:
+    """Resolve and fetch RubyGems dependencies."""
+    components = []
+    environment_variables = [
+        EnvironmentVariable(name="BUNDLE_CACHE_ALL", value="true"),
+        EnvironmentVariable(name="BUNDLE_CACHE_PATH", value="${output_dir}/deps/rubygems"),
+        EnvironmentVariable(name="BUNDLE_FORCE_RUBY_PLATFORM", value="true"),
```

Comment:
I'm a little confused if these variables are really relevant for the build phase. I was able to successfully build the [test repo](https://github.com/slimreaper35/cachi2-rubygems) you set up and the only thing I really needed was to properly set `BUNDLE_CACHE_PATH`.

By teading the bundler docs, I got the impression that those variables would be relevant in case we'd use bundler for doing the prefetch, which is not the case.

Reply:
Yeah, I will remove `BUNDLE_CACHE_ALL`. Checking the ruby platform is currently done by gemlock-parser. Do we want to set that during the build as well ?